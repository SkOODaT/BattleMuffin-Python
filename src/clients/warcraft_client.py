import requests
from authlib.integrations.requests_client import OAuth2Session
from uplink import Consumer, Query, get, params, returns

from config.client_config import ClientConfiguration
from config.region_config import Locale, Region


class WarcraftClient(Consumer):
    def __init__(self, client_id: str, client_secret: str, region: Region = Region.us, locale: Locale = Locale.en_US):
        self.client_config = ClientConfiguration(client_id, client_secret, region, locale)

        oath_response = requests.get(f"{self.client_config.oauth_host}/oauth/.well-known/openid-configuration").json()
        token_endpoint = oath_response['token_endpoint']
        session = OAuth2Session(client_id, client_secret, scope="openid", token_endpoint=token_endpoint, grant_type="client_credentials")
        session.fetch_token()
        super().__init__(base_url=self.client_config.host, client=session)
        self._inject(Query("locale").with_value(self.client_config.locale.name))

    @returns.json
    @params({"namespace": "static-us"})
    @get("/data/wow/achievement-category/index")
    def get_achievement_category_index(self):
        """Returns an index of achievement categories."""
