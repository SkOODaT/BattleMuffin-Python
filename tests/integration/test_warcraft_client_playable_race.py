import os
from battlemuffin.clients.warcraft_client import WarcraftClient


def test_get_playable_races_index(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_races_index()
    assert response == snapshot


def test_get_playable_race(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_race(2)
    assert response == snapshot
