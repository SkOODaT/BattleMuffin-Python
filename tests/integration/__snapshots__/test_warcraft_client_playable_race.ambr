# name: test_get_playable_race
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-race/2?namespace=static-8.3.0_32861-us',
      },
    },
    'faction': <class 'dict'> {
      'name': 'Horde',
      'type': 'HORDE',
    },
    'gender_name': <class 'dict'> {
      'female': 'Orc',
      'male': 'Orc',
    },
    'id': 2,
    'is_allied_race': False,
    'is_selectable': True,
    'name': 'Orc',
  }
---
# name: test_get_playable_races_index
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-race/?namespace=static-8.3.0_32861-us',
      },
    },
    'races': <class 'list'> [
      <class 'dict'> {
        'id': 1,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/1?namespace=static-8.3.0_32861-us',
        },
        'name': 'Human',
      },
      <class 'dict'> {
        'id': 2,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/2?namespace=static-8.3.0_32861-us',
        },
        'name': 'Orc',
      },
      <class 'dict'> {
        'id': 3,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/3?namespace=static-8.3.0_32861-us',
        },
        'name': 'Dwarf',
      },
      <class 'dict'> {
        'id': 4,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/4?namespace=static-8.3.0_32861-us',
        },
        'name': 'Night Elf',
      },
      <class 'dict'> {
        'id': 5,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/5?namespace=static-8.3.0_32861-us',
        },
        'name': 'Undead',
      },
      <class 'dict'> {
        'id': 6,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/6?namespace=static-8.3.0_32861-us',
        },
        'name': 'Tauren',
      },
      <class 'dict'> {
        'id': 7,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/7?namespace=static-8.3.0_32861-us',
        },
        'name': 'Gnome',
      },
      <class 'dict'> {
        'id': 8,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/8?namespace=static-8.3.0_32861-us',
        },
        'name': 'Troll',
      },
      <class 'dict'> {
        'id': 9,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/9?namespace=static-8.3.0_32861-us',
        },
        'name': 'Goblin',
      },
      <class 'dict'> {
        'id': 10,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/10?namespace=static-8.3.0_32861-us',
        },
        'name': 'Blood Elf',
      },
      <class 'dict'> {
        'id': 11,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/11?namespace=static-8.3.0_32861-us',
        },
        'name': 'Draenei',
      },
      <class 'dict'> {
        'id': 22,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/22?namespace=static-8.3.0_32861-us',
        },
        'name': 'Worgen',
      },
      <class 'dict'> {
        'id': 24,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/24?namespace=static-8.3.0_32861-us',
        },
        'name': 'Pandaren',
      },
      <class 'dict'> {
        'id': 25,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/25?namespace=static-8.3.0_32861-us',
        },
        'name': 'Pandaren',
      },
      <class 'dict'> {
        'id': 26,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/26?namespace=static-8.3.0_32861-us',
        },
        'name': 'Pandaren',
      },
      <class 'dict'> {
        'id': 27,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/27?namespace=static-8.3.0_32861-us',
        },
        'name': 'Nightborne',
      },
      <class 'dict'> {
        'id': 28,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/28?namespace=static-8.3.0_32861-us',
        },
        'name': 'Highmountain Tauren',
      },
      <class 'dict'> {
        'id': 29,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/29?namespace=static-8.3.0_32861-us',
        },
        'name': 'Void Elf',
      },
      <class 'dict'> {
        'id': 30,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/30?namespace=static-8.3.0_32861-us',
        },
        'name': 'Lightforged Draenei',
      },
      <class 'dict'> {
        'id': 31,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/31?namespace=static-8.3.0_32861-us',
        },
        'name': 'Zandalari Troll',
      },
      <class 'dict'> {
        'id': 32,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/32?namespace=static-8.3.0_32861-us',
        },
        'name': 'Kul Tiran',
      },
      <class 'dict'> {
        'id': 34,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/34?namespace=static-8.3.0_32861-us',
        },
        'name': 'Dark Iron Dwarf',
      },
      <class 'dict'> {
        'id': 35,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/35?namespace=static-8.3.0_32861-us',
        },
        'name': 'Vulpera',
      },
      <class 'dict'> {
        'id': 36,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/36?namespace=static-8.3.0_32861-us',
        },
        'name': "Mag'har Orc",
      },
      <class 'dict'> {
        'id': 37,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-race/37?namespace=static-8.3.0_32861-us',
        },
        'name': 'Mechagnome',
      },
    ],
  }
---
